documentation/modules/oauth/con-oauth-reauthentication.adoc

```diff
@@ -1,23 +1,41 @@
 
-When the re-authentication mechanism is enabled in the configuration of an `oauth` type listener, the broker's authenticated session expires when the access token expires.
-The client must then re-authenticate to the existing session by sending a new, valid access token to the broker, without dropping the connection.
+Session re-authentication is disabled by default. 
+To enable and configure it, you set a time value for `maxSecondsWithoutReauthentication` in the `oauth` listener configuration in the `Kafka` custom resource. 
+The same property is used to configure session re-authentication for OAUTHBEARER and PLAIN authentication. 
+For an example configuration, see xref:proc-oauth-authentication-broker-config-{context}[].
+
+Session re-authentication can be used with _fast local JWT_ or _introspection endpoint_ token validation.
+
+.Client re-authentication
+
+When the broker's authenticated session expires, the client must re-authenticate to the existing session by sending a new, valid access token to the broker, without dropping the connection.
 
 If token validation is successful, a new client session is started using the existing connection.
 If the client fails to re-authenticate, the broker will close the connection if further attempts are made to send or receive messages.
 Java clients that use Kafka client library 2.2 or later automatically re-authenticate if the re-authentication mechanism is enabled on the broker.
 
-You enable session re-authentication in the `Kafka` resource. Set the `maxSecondsWithoutReauthentication` property for a TLS listener with `type: oauth` authentication.
-Session re-authentication is supported for both types of token validation (fast local JWT and introspection endpoint). For an example configuration, see xref:proc-oauth-authentication-broker-config-{context}[].
+.Differences in session expiry for OAUTHBEARER and PLAIN
+
+There are some differences in how session expiry works when session re-authentication is configured for OAUTHBEARER and PLAIN authentication.
 
-For more information about the re-authentication mechanism, which was added in Kafka version 2.2, see link:https://cwiki.apache.org/confluence/display/KAFKA/KIP-368%3A+Allow+SASL+Connections+to+Periodically+Re-Authenticate[KIP-368^].
+* For OAUTHBEARER, the broker's authenticated session will expire at the configured `maxSecondsWithoutReauthentication`. 
+The session will expire earlier if the access token expires before the `maxSecondsWithoutReauthentication`. 
+
+* For PLAIN, the broker's authenticated session will expire at the configured `maxSecondsWithoutReauthentication`. 
+
+If `maxSecondsWithoutReauthentication` is not configured, OAUTHBEARER and PLAIN clients can remain connected to brokers indefinitely, without needing to re-authenticate. 
+In that case, the sessions do not end with access token expiry, but this can be considered when configuring proper authorization. 
```

Comment:
```suggestion
In this case, the sessions do not end with access token expiry, but this can be considered when configuring proper authorization. 
```

Reply:
Not sure what _but this can be considered when configuring proper authorization._ is trying to say. Or what _proper authorization_ might be.

Reply:
I may have distorted the meaning. @mstruk - Please can you clarify?

Updated to:

> Authenticated sessions do not end with access token expiry. 
> However, this can be considered when configuring authorization, for example, by using `keycloak` authorization or installing a custom authorizer.